bots/example_bots/gridbot.py

```diff
@@ -232,6 +200,56 @@ def cancel_all_active_orders(self) -> None:
     # helper methods
     ######################################################################
 
+    def _amount_in_market(self, side: OrderSide) -> Decimal:
+        active_orders = list(filter(lambda order: order.order_side == side, self.active_limit_orders.values()))
+        pending_orders = []
+        for pending_transaction in self.pending_transactions.values():
+            pending_orders.extend(list(filter(lambda order: order.order_side == side, pending_transaction.orders)))
+
+        amount = (
+            sum(map(lambda order: order.remaining_size(), active_orders)) +
+            sum(map(lambda order: order.size, pending_orders))
+        )
+
+        return amount
+
+    def _check_sufficient_inventory_to_place(self, new_orders: List[NewLimitOrder]) -> List[NewLimitOrder]:
```

Comment:
one thing i have been thinking throughout, we probably should be handling token values throughout the bot as their `integer` representation instead of their `decimal` value. we can always convert to a decimal value when we need it in human readable format, or store both on the `integer` and `decimal` value in the same object

Reply:
I'm not sure this is necessary. The purpose of the Decimal class is to preserve precision (by default to `28` places: https://docs.python.org/3/library/decimal.html). 

I think this is sufficiently accurate and I think the issues users are experiencing are due to a combination of:

1. Protocol rounding
2. User error

However I am happy to discuss this. @denverbaumgartner  @bghughes